### PR TITLE
JSON API: avoid notices when updating a post in the WordPress.com post editor

### PR DESCRIPTION
**Pull request by jeherve**
Here were the notices I got:

`[11-Jul-2015 15:43:32] PHP Notice:  Undefined index:  menu_order in /wp-content/plugins/jetpack/json-endpoints/class.wpcom-json-api-update-post-v1-1-endpoint.php on line 232
[11-Jul-2015 15:43:32] PHP Notice:  Undefined index:  metadata in /wp-content/plugins/jetpack/json-endpoints/class.wpcom-json-api-update-post-v1-1-endpoint.php on line 245
[11-Jul-2015 15:43:32] PHP Notice:  Undefined index:  client_id in /wp-content/plugins/jetpack/json-endpoints/class.wpcom-json-api-update-post-v1-1-endpoint.php on line 401
[11-Jul-2015 15:43:32] PHP Notice:  Undefined index:  type in /wp-content/plugins/jetpack/json-endpoints/class.wpcom-json-api-update-post-v1-1-endpoint.php on line 558
`

### CODE REVIEW

File: json-endpoints/class.wpcom-json-api-update-post-endpoint.php [hunk 1]
```diff
@@ -236,17 +244,25 @@ function write_post( $path, $blog_id, $post_id ) {
...
-		$likes = $input['likes_enabled'];
-		$sharing = $input['sharing_enabled'];
+		if ( isset( $input['likes_enabled'] ) ) {
+			$likes = $input['likes_enabled'];
+			unset( $input['likes_enabled'] );
+		}
 
-		unset( $input['likes_enabled'] );
-		unset( $input['sharing_enabled'] );
+		if ( isset( $input['likes_enabled'] ) ) {
```
Comment (zinigor) [inline json-endpoints/class.wpcom-json-api-update-post-endpoint.php]: I think this should be `sharing_enabled`.

**Comment by zinigor** [comment]
Read the code, looks fine except for one small thing, after fixing feel free to tag as `Review Complete`.

File: json-endpoints/class.wpcom-json-api-update-post-endpoint.php (hunk 1 above)
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-update-post-endpoint.php]: Thanks for catching that one! Fixed in e17faab74d15e4c3a888409151163a7a25306a74

**Comment by dereksmart** [comment]
isset() all the things!

**Pull request merged**